src/schemaCompare/schemaCompareWebViewController.ts

```diff
@@ -0,0 +1,323 @@
+        sourceContext: any,
+    ): Promise<mssql.SchemaCompareEndpointInfo> {
+        let ownerUri =
+            await this.connectionMgr.getUriForConnection(connectionProfile);
+        let user = connectionProfile.user;
+        if (!user) {
+            user = "default";
+        }
+
+        const source = {
+            endpointType: mssql.SchemaCompareEndpointType.Database,
+            serverDisplayName: `${connectionProfile.server} (${user})`,
+            serverName: connectionProfile.server,
+            databaseName: ObjectExplorerUtils.getDatabaseName(sourceContext),
+            ownerUri: ownerUri,
+            packageFilePath: "",
+            connectionDetails: undefined,
+            connectionName: connectionProfile.profileName
+                ? connectionProfile.profileName
+                : "",
+            projectFilePath: "",
+            targetScripts: [],
+            dataSchemaProvider: "",
+            extractTarget: mssql.ExtractTarget.schemaObjectType,
+        };
+
+        return source;
+    }
+
+    private getEndpointInfoFromDacpac(
+        sourceDacpac: string,
+    ): mssql.SchemaCompareEndpointInfo {
+        const source = {
+            endpointType: mssql.SchemaCompareEndpointType.Dacpac,
+            serverDisplayName: "",
+            serverName: "",
+            databaseName: "",
+            ownerUri: "",
+            packageFilePath: sourceDacpac,
+            connectionDetails: undefined,
+            projectFilePath: "",
+            targetScripts: [],
+            dataSchemaProvider: "",
+            extractTarget: mssql.ExtractTarget.schemaObjectType,
+        };
+
+        return source;
+    }
+
+    private getEndpointInfoFromProject(
+        sourceProject: string,
+    ): mssql.SchemaCompareEndpointInfo {
+        const source = {
+            endpointType: mssql.SchemaCompareEndpointType.Project,
+            packageFilePath: "",
+            serverDisplayName: "",
+            serverName: "",
+            databaseName: "",
+            ownerUri: "",
+            connectionDetails: undefined,
+            projectFilePath: sourceProject,
+            targetScripts: [],
+            dataSchemaProvider: undefined,
+            extractTarget: mssql.ExtractTarget.schemaObjectType,
+        };
+
+        return source;
+    }
+
+    private isTreeNodeInfoType(node: any): boolean {
+        if (node instanceof TreeNodeInfo) {
+            return true;
+        }
+
+        return false;
+    }
+
+    private getFullSqlProjectsPathFromNode(node: any): string {
+        return node.treeDataProvider?.roots[0]?.projectFileUri?.fsPath ?? "";
+    }
+
+    private registerRpcHandlers(): void {
+        this.registerReducer("compare", async (state, payload) => {
+            const result = await compare(
+                this.operationId,
+                payload,
+                this.schemaCompareService,
+            );
+
+            return { ...state, schemaCompareResult: result };
```

Comment:
why not just set `state.schemaCompareResult = result` then return it?   Same below if applicable

Reply:
That works too. I'll change it.